xbmc/Application.cpp

```diff
@@ -1897,7 +1897,10 @@ void CApplication::Render()
     g_Windowing.FinishPipeline();
   }
 
-  g_graphicsContext.Flip(hasRendered);
+  // TODO: find better solution
+  // if video is rendered to a separate layer, we should not block this thread
+  if (!m_pPlayer->IsRenderingVideoLayer() || hasRendered)
+    g_graphicsContext.Flip(hasRendered);
 
```

Comment:
The current code is correct. We call Flip() if gui has rendered or video is rendered to gui layer. bypass renderer need to throttle the renderloop themselves. Applications can't anymore because it is not aware of video fps.